### PR TITLE
javadoc - Adding default constructor and javadoc for :ethereum:api

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
javadoc - Adding default constructor and javadoc for :ethereum:api (~200 files)

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [x] acceptance tests: `./gradlew acceptanceTest`
- [x] integration tests: `./gradlew integrationTest`
- [x] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadBodiesResultV1.java [hunk 1]
```diff
@@ -28,28 +28,46 @@
 import com.fasterxml.jackson.annotation.JsonValue;
 import org.apache.tuweni.bytes.Bytes;
 
+/** The type Engine get payload bodies result v 1. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadBodiesResultV1.java:31]:
```suggestion
/** The type Engine get payload bodies result v1. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadBodiesResultV1.java [hunk 2]
```diff
@@ -28,28 +28,46 @@
...
 @JsonPropertyOrder({"payloadBodies"})
 public class EngineGetPayloadBodiesResultV1 {
 
   private final List<PayloadBody> payloadBodies;
 
+  /** Instantiates a new Engine get payload bodies result v 1. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadBodiesResultV1.java:37]:
```suggestion
  /** Instantiates a new Engine get payload bodies result v1. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadBodiesResultV1.java [hunk 3]
```diff
@@ -28,28 +28,46 @@
...
   public EngineGetPayloadBodiesResultV1() {
     this.payloadBodies = Collections.<PayloadBody>emptyList();
   }
 
+  /**
+   * Instantiates a new Engine get payload bodies result v 1.
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadBodiesResultV1.java:43]:
```suggestion
   * Instantiates a new Engine get payload bodies result v1.
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/Web3JsonRpcMethods.java [hunk 4]
```diff
@@ -21,10 +21,16 @@
 
 import java.util.Map;
 
+/** The type Web 3 json rpc methods. */
 public class Web3JsonRpcMethods extends ApiGroupJsonRpcMethods {
 
   private final String clientVersion;
 
+  /**
+   * Instantiates a new Web 3 json rpc methods.
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/Web3JsonRpcMethods.java:30]:
```suggestion
   * Instantiates a new Web3 json rpc methods.
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmOperationExecutionReport.java [hunk 5]
```diff
@@ -24,43 +24,89 @@ public class VmOperationExecutionReport {
   private Store store;
   private long used;
 
+  /** Instantiates a new Vm operation execution report. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmOperationExecutionReport.java:27]:
```suggestion
  /** Instantiates a new VM operation execution report. */
```

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmOperationExecutionReport.java:27]: this one is in a few spots

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 6]
```diff
@@ -21,213 +21,345 @@
...
   // Miner failures
   COINBASE_NOT_SET(-32010, "Coinbase not set. Unable to start mining without a coinbase"),
+  /** The No hashes per second. */
   NO_HASHES_PER_SECOND(-32011, "No hashes being generated by the current node"),
+  /** The Target gas limit modification unsupported. */
   TARGET_GAS_LIMIT_MODIFICATION_UNSUPPORTED(
       -32011, "The node is not attempting to target a gas limit so the target can't be changed"),
 
+  /** The Coinbase not specified. */
   // Wallet errors
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:160]:
```suggestion
  // coinbase errors
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/DepositParameter.java [hunk 7]
```diff
@@ -61,6 +77,11 @@ public static DepositParameter fromDeposit(final Deposit deposit) {
         deposit.getIndex().toBytes().toQuantityHexString());
   }
 
+  /**
+   * To deposit deposit.
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/DepositParameter.java:81]:
```suggestion
   * To deposit.
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/DepositParameter.java [hunk 8]
```diff
@@ -52,6 +62,12 @@ public DepositParameter(
     this.index = index;
   }
 
+  /**
+   * From deposit deposit parameter.
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/DepositParameter.java:66]:
```suggestion
   * Deposit parameter from deposit.
```

**Review by macfarla — APPROVED** [review]
few nits as suggestions, otherwise LGTM

**Comment by siladu** [comment]
Is it possible to prevent the tool from turning the classname into a "sentence"?

**Comment by usmansaleem** [comment]
Will recreate with smaller PRs.

**Pull request closed without merging**